hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -1152,12 +1153,19 @@ def example_deletion_with_block_lowering(self, block, ex):
         del buf[ex.start : ex.end]
         self.incorporate_new_buffer(buf)
 
+    @derived_value
+    def groups_for_reordering(self):
+        result = defaultdict(list)
+        for e in self.examples:
+            result[(e.parent, e.label)].append(e)
```

Comment:
Does this means that we only attempt to reorder examples with the same label *and parent*, and if so why?  My intuition is that grouping by label only will be faster enough and somewhat more powerful.

Reply:
It does, but that was also the case before hand! Experience so far of trying to make it work suggests that grouping by label only is *not* faster enough. I think there's probably a way to make it work but it needs more investigation.